addons/purchase_edi_ubl_bis3/models/purchase_edi_xml_ubl_bis3.py

```diff
@@ -0,0 +1,245 @@
+from lxml import etree
+from typing import NamedTuple
+
+from odoo import models, _
+from odoo.tools import html2plaintext, cleanup_xml_node
+
+
+class LineAdapter(NamedTuple):
+    """
+    This named tuple was introduced to make _get_uom_unece_code() compatible with purchase order lines
+    TODO: remove it when we fix the signature of _get_uom_unece_code() to accept uom instead of line
+    """
+    product_uom_id: models.Model
+
+
+class PurchaseEdiXmlUBLBIS3(models.AbstractModel):
+    _name = "purchase.edi.xml.ubl_bis3"
+    _description = "UBL BIS 3 Peppol Order transaction 3.4"
+
+    # -------------------------------------------------------------------------
+    # EXPORT
+    # -------------------------------------------------------------------------
+
+    def _export_purchase_order_filename(self, purchase_order):
+        return f"{purchase_order.name.replace('/', '_')}_ubl_bis3.xml"
+
+    def _get_country_vals(self, country):
+        return {
+            'country': country,
+
+            'identification_code': country.code,
+            'name': country.name,
+        }
+
+    def _get_partner_address_vals(self, partner):
+        return {
+            'street_name': partner.street,
+            'additional_street_name': partner.street2,
+            'city_name': partner.city,
+            'postal_zone': partner.zip,
+            'country_subentity': partner.state_id.name,
+            'country_identification_code': partner.country_id.code
+        }
+
+    def _get_partner_party_tax_scheme_vals(self, partner):
+        return {
+            'company_id': partner.vat,
+            'tax_scheme_vals': {'id': 'VAT'},
+        }
+
+    def _get_partner_party_legal_entity_vals(self, partner):
+        return {
+            'registration_name': partner.name,
+            'company_id': partner.vat,
+            'registration_address_vals': self._get_partner_address_vals(partner),
+        }
+
+    def _get_partner_contact_vals(self, partner):
+        return {
+            'name': partner.name,
+            'telephone': partner.phone or partner.mobile,
+            'electronic_mail': partner.email,
+        }
+
+    def _get_partner_party_vals(self, partner, role):
+        vals = {
+            'party_name': partner.display_name,
+            'postal_address_vals': self._get_partner_address_vals(partner),
+            'contact_vals': self._get_partner_contact_vals(partner),
+        }
+        if role == 'customer':
+            vals['party_tax_scheme_vals'] = self._get_partner_party_tax_scheme_vals(partner.commercial_partner_id)
+        return vals
+
+    def _get_delivery_party_vals(self, order):
+        partner = order.dest_address_id or order.partner_id
```

Comment:
it delivery to our warehouse, are you sure, you have to take the partner on the order and not warehouse or company delivery address

Reply:
It delivers to a dropshipping address if one exists, otherwise it delivers to our own warehouse. This is because it's a purchase order.